Data/ByteString.hs

```diff
@@ -1413,8 +1413,15 @@ findSubstring pat src
 
 {-# DEPRECATED findSubstring "findSubstring is deprecated in favour of breakSubstring." #-}
 
--- | Find the indexes of all (possibly overlapping) occurences of a
--- substring in a string.
+-- | Find the indices of all non-overlapping occurences of a substring in a
+-- string.
+--
+-- Note, prior to @0.10.6.0@, this function returned the indices of all
+-- possibly-overlapping matches.  The behaviour change appears to be
+-- inadvertent, but the function had already been deprecated for more than
+-- seven years.  At this time (@0.10.10.1@), the deprecation was twelve years
+-- in the past.  Given the deprecation and unnoticed behaviour change, it will
+-- likely be removed in @0.11.*@.
```

Comment:
What is this comment based on? Is there a corresponding issue?

Reply:
Adding more detail.

Reply:
I would probably avoid promising to remove this function in a specific future version. 

(My **personal** preference is to leave only the first sentence of the paragraph "Note, prior to...". The reader does not neccessarily care whether this change was adverent or not and how long ago it happened. This info may be put below, for the attention of future maintainers, but outside of the haddock comment.)